pydm/widgets/template_repeater.py

```diff
@@ -284,13 +289,13 @@ def dataSource(self, data_source):
             self._data_source = data_source
             if self._data_source:
                 try:
-                    # Expand user (~ or ~user) and environment variables.
-                    fname = os.path.expanduser(os.path.expandvars(self._data_source))
-                    if is_pydm_app():
-                        # If we're running this inside the PyDM app, we can
-                        # make sure the path is relative to the currently loaded
-                        # display (.ui or .py file).
-                        fname = self.app.get_path(fname)
+                    parent_display = self.find_parent_display()
+                    base_path = None
+                    if parent_display:
+                        base_path = os.path.dirname(
+                            parent_display.loaded_file())
+                    fname = find_file(self._data_source, base_path=base_path)
+
                     with open(fname) as f:
```

Comment:
Avoids errors in Qt Designer, when `dataSource` is being typed in.
Avoids exceptions in general, preferring the `logger.error` at runtime.
```suggestion
                    if not fname:
                        if not is_qt_designer():
                            logger.error('Cannot locate data source file for PyDMTemplateRepeater.')
                        self.data = []
                    else:
                        with open(fname) as f:
                            try:
                                self.data = json.load(f)
                            except ValueError:
                                logger.error('Failed to parse data source file for PyDMTemplateRepeater.')
                                self.data = []
```